components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1677,4 +1768,39 @@ private static String hexify(byte bytes[]) {
         return essentailClaimslist;
     }
 
+    public static String getSanitizedUserStoreDomain(String userStoreDomain) {
+        if (userStoreDomain != null) {
+            userStoreDomain = userStoreDomain.toUpperCase();
+        } else {
+            userStoreDomain = IdentityUtil.getPrimaryDomainName();
+        }
+        return userStoreDomain;
+    }
+
+    public static String getUserStoreForFederatedUser(AuthenticatedUser authenticatedUser) throws
+            IdentityOAuth2Exception {
+
+        String userStoreDomain = OAuth2Util.getUserStoreDomainFromUserId(authenticatedUser.toString());
+        if (!OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && authenticatedUser.
+                isFederatedUser()) {
+            userStoreDomain = OAuth2Util.getFederatedUserDomain(authenticatedUser.getFederatedIdPName());
+        }
+        return userStoreDomain;
+    }
+
+    public static String addUsernameToToken(AuthenticatedUser authenticatedUser, String token) {
+
+        String usernameForToken = authenticatedUser.toString();
```

Comment:
Should we do a null check for authenticatedUser or will it be handled at a layer above this?